src/toxcore/dht/server/mod.rs

```diff
@@ -1174,7 +1174,13 @@ impl Server {
         let onion_symmetric_key = self.onion_symmetric_key.read();
         let payload = packet.onion_return.get_payload(&onion_symmetric_key);
         let payload = match payload {
-            Err(e) => return Box::new(future::err(e)),
+            Err(e) => {
+                // Onion symmetric key is changed every 2 hours to enforce onion
+                // paths expiration. It means that we can get packets with old
+                // onion key. So we do not consider this as error.
+                trace!("Failed to decrypt onion_return: {}", e);
```

Comment:
Added, but later if we switch to enums for errors we will probably want one unified error type for these errors, because it's the same error, just happens in different places.